In [7]:
import pandas as pd
import numpy as np
import seaborn as sns #Serve per fare i grafici
import matplotlib.pyplot as plt
import os
os.getcwd()
os.listdir(r"C:\Users\Gianmario\Desktop\Python\Esercitazioni\Corso_GiGroup\Palleschi_DepositoCorso\Lezione_13")





['Cars Datasets 2025.csv', 'EDA_macchine.ipynb']

In [12]:
df = pd.read_csv(
    r"C:\Users\Gianmario\Desktop\Python\Esercitazioni\Corso_GiGroup\Palleschi_DepositoCorso\Lezione_13\Cars Datasets 2025.csv",
    encoding="cp1252"
)

print("Prime 5 righe")
# Mostra le prime 5 osservazioni del dataset, utilissimo per verificare che sia stato caricato correttamente
df.head()




Prime 5 righe


Company Names     Cars Names      Engines CC/Battery Capacity HorsePower  \
0       FERRARI  SF90 STRADALE           V8             3990 cc     963 hp   
1   ROLLS ROYCE        PHANTOM          V12             6749 cc     563 hp   
2          Ford            KA+  1.2L Petrol            1,200 cc   70-85 hp   
3      MERCEDES        GT 63 S           V8            3,982 cc     630 hp   
4          AUDI     AUDI R8 Gt          V10            5,204 cc     602 hp   

  Total Speed Performance(0 - 100 )KM/H      Cars Prices       Fuel Types  \
0    340 km/h                   2.5 sec      $1,100,000   plug in hyrbrid   
1    250 km/h                   5.3 sec        $460,000            Petrol   
2    165 km/h                  10.5 sec  $12,000-$15,000           Petrol   
3    250 km/h                   3.2 sec        $161,000            Petrol   
4    320 km/h                   3.6 sec        $253,290            Petrol   

  Seats        Torque  
0     2        800 Nm  
1     5        900 Nm  
2     5  100 - 140 Nm  
3     4        900 Nm  
4     2        560 Nm

In [13]:
print("Ultime 5 righe")
# Mostra le ultime 5 osservazioni, utile per controllare eventuali anomalie in fondo al file
df.tail()



Ultime 5 righe


Company Names         Cars Names                             Engines  \
1213        Toyota       Crown Signia                      2.5L Hybrid I4   
1214        Toyota  4Runner (6th Gen)  2.4L Turbo I4 (i-FORCE MAX Hybrid)   
1215        Toyota      Corolla Cross              2.0L Gas / 2.0L Hybrid   
1216        Toyota             C-HR+                   1.8L / 2.0L Hybrid   
1217        Toyota     RAV4 (6th Gen)        2.5L Hybrid / Plug-in Hybrid   

        CC/Battery Capacity    HorsePower Total Speed  \
1213                2487 cc        240 hp    180 km/h   
1214      2393 cc + Battery        326 hp    180 km/h   
1215  1987 cc / Hybrid batt  169 – 196 hp    190 km/h   
1216  1798 / 1987 cc + batt  140 – 198 hp    180 km/h   
1217         2487 cc + batt  219 – 302 hp    200 km/h   

     Performance(0 - 100 )KM/H        Cars Prices               Fuel Types  \
1213                   7.6 sec  $43,590 – $48,000  Hybrid (Gas + Electric)   
1214                   6.8 sec           $50,000                    Hybrid   
1215             8.0 – 9.2 sec  $25,210 – $29,135             Gas / Hybrid   
1216            7.9 – 10.5 sec           € 33,000                   Hybrid   
1217             6.0 – 8.1 sec  $29,000 – $43,000         Hybrid / Plug-in   

     Seats        Torque  
1213     5        239 Nm  
1214     7        630 Nm  
1215     5  190 – 210 Nm  
1216     5  190 – 205 Nm  
1217     5  221 – 400 Nm

In [14]:
# df.shape è una proprietà, non una funzione → NON va usato con le parentesi
df.shape
# Restituisce una tupla (numero_righe, numero_colonne)



(1218, 11)

In [15]:
# Fornisce informazioni sulla struttura del dataframe:
# - nomi delle variabili
# - tipo di dati (int, float, object...)
# - quanti valori non null per colonna
df.info()



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1218 entries, 0 to 1217
Data columns (total 11 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   Company Names              1218 non-null   object
 1   Cars Names                 1218 non-null   object
 2   Engines                    1218 non-null   object
 3   CC/Battery Capacity        1215 non-null   object
 4   HorsePower                 1218 non-null   object
 5   Total Speed                1218 non-null   object
 6   Performance(0 - 100 )KM/H  1212 non-null   object
 7   Cars Prices                1218 non-null   object
 8   Fuel Types                 1218 non-null   object
 9   Seats                      1218 non-null   object
 10  Torque                     1217 non-null   object
dtypes: object(11)
memory usage: 104.8+ KB


In [ ]:
# Restituisce statistiche descrittive sulle variabili numeriche:
# - count (numero di valori)
# - mean (media)
# - std (deviazione standard)
# - min, max
# - quartili 25%, 50%, 75%
df.describe()



Company Names       Cars Names Engines CC/Battery Capacity HorsePower  \
count           1218             1218    1218                1215       1218   
unique            37             1201     356                 311        456   
top           Nissan  Polo BlueMotion      I4             1984 cc     355 hp   
freq             149                2      64                  31         23   

       Total Speed Performance(0 - 100 )KM/H Cars Prices Fuel Types Seats  \
count         1218                      1212        1218       1218  1218   
unique         114                       180         535         23    19   
top       250 km/h                   6.5 sec    $35,000      Petrol     5   
freq           145                        45          36        871   692   

        Torque  
count     1217  
unique     263  
top     400 Nm  
freq        72

Quando usi df.describe() su questo dataset pandas riconosce tutte le colonne come oggetti (stringhe), quindi produce automaticamente le statistiche descrittive per variabili categoriche, non per variabili numeriche.

Per questo vedi:

count → numero di valori non nulli

unique → numero di valori diversi

top → valore più frequente

freq → frequenza del valore più comune

Queste sono le statistiche standard per variabili di tipo object.
Tutte e 11 le variabili sono lette come object, anche quelle che dovrebbero essere numeriche (HorsePower, Speed, Prices, Torque ecc.).

Quindi df.describe() interpreta tutto come testo e ti mostra statistiche per categorie.
Perché succede?

Perché nel tuo CSV:

i numeri probabilmente contengono unità di misura (es. “355 hp”, “250 km/h”, “400 Nm”)

o simboli come $, km/h, cc, hp, Nm, sec

il separatore dei decimali non è standard

oppure i valori sono misti (testo + numeri)

Quindi pandas NON può convertirli automaticamente in numerici.


Come ottenere statistiche numeriche vere?

Devi trasformare in numerico le colonne che sono numeriche “nascoste”, ad esempio: